### PR TITLE
Add control input dependency to Java bindings

### PR DESCRIPTION
**Pull request by kbsriram**
Add a builder method to OperationBuilder to allow adding
control inputs to Operations.

### CODE REVIEW

File: tensorflow/java/src/test/java/org/tensorflow/OperationBuilderTest.java [hunk 1]
```diff
@@ -149,6 +151,36 @@ public void setAttrShape() {
     }
   }
 
+  @Test
+  public void addControlInput() {
+    assertEquals(0, lessEqualControl(2, 3).size());
+    try {
+      lessEqualControl(3, 2);
```
Comment (asimshankar) [inline tensorflow/java/src/test/java/org/tensorflow/OperationBuilderTest.java]: There is a lot happening in lessEqualControl to be able to tie the exception to the specific line we want.
It might be better to create a simpler graph with a placeholder and have the `try{}` block around just the session run call. Something like:

```java
public void addControlInput() {
  try (Graph g = new Graph();
         Session s = new Session(g);
         Tensor yes = Tensor.create(true);
         Tensor no = Tensor.create(false)) {
    Output placeholder = TestUtil.placeholder(g, "boolean", DataType.BOOL);
    Output assert = g.opBuilder("Assert", "assert")
       .addInput(placeholder)
       .addInputList(new Output[]{})
       .build();
      Output noop = g.opBuilder("NoOp", "noop")
          .addControlInput(assert)
          .build();

     // No problems when the assert succeeds
     s.runner().feed(placeholder, yes).addTarget(noop).run();

     // Exception thrown by the execution of the Assert node
     try {
        s.runner().feed(placeholder, no).addTarget(noop).run();
        fail(...);
      } catch (IllegalArgumentException e) {
        // expected
      }
  }
}
```

File: tensorflow/java/src/test/java/org/tensorflow/OperationBuilderTest.java [hunk 2]
```diff
@@ -24,6 +24,8 @@
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
+import java.util.List;
```
Comment (asimshankar) [inline tensorflow/java/src/test/java/org/tensorflow/OperationBuilderTest.java]: Why was this import needed?

**Review by asimshankar — COMMENTED** [review]
Thanks very much for the change.

Some minor comments, otherwise looks great.

**Review by asimshankar — APPROVED** [review]
Thanks!

File: tensorflow/java/src/test/java/org/tensorflow/OperationBuilderTest.java (hunk 1 above)
Comment (kbsriram) [inline tensorflow/java/src/test/java/org/tensorflow/OperationBuilderTest.java]: Nice! Thanks @asimshankar for the pointers to simplify the test, and the other suggestions. I've incorporated all of them in the next update of this PR.

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Review by asimshankar — APPROVED** [review]

**Comment by kbsriram** [comment]
Is there perhaps a flaky test in the windows CMake build?

The only time the ci build ran it reported an error in a seemingly [unrelated area](https://ci.tensorflow.org/job/tensorflow-pr-win-cmake-py/2249/console) of the code; and I'm not able to connect the dots between the reported failure and this change.

**Comment by caisq** [comment]
@tensorflow-jenkins test this please
